content/200-concepts/100-components/02-prisma-client/000-working-with-prismaclient/115-connection-pool.mdx

```diff
@@ -0,0 +1,84 @@
+---
+title: Connection pool
+metaDescription: Prisma's query engine creates a connection pool to store and manage database connections.
+tocDepth: 2
+---
+
+<TopBlock>
+
+The query engine manages a **connection pool** of database connections. The pool is created when Prisma Client opens the _first_ connection to the database, which can happen in one of two ways:
+
+- By [explicitly calling `$connect()`](connection-management#connect) _or_
+- By running the first query, which calls `$connect()` under the hood
+
+The connection pool has a **connection limit** and a **pool timeout**, which are controlled by connection string parameters.
+
+</TopBlock>
+
+## How the connection pool works
+
+The following steps describe how the query engine uses the connection pool:
+
+1. The query engine instantiates a connection pool with a [congfigurable pool size](#setting-the-connection-limit) and [pool timeout](#setting-the-connection-pool-timeout).
+1. The query engine reserves connections from the pool and processes incoming queries.
```

Comment:
This might be missing that it also opens additional connection to the database until the pool size is reached and the pool is "full".

Reply:
@janpio Like so?

Reply:
(Ignore greyed out 'commit suggestion' option - the content in green is what's in the branch now)

Reply:
Uhhh, that "suggested change" in my comment really confused me - as it is not correct. I'll open a new comment on these lines with a new suggestion.